.github/workflows/node-ci.yml

```diff
@@ -228,6 +229,111 @@ jobs:
+                  pnpm tsc --declaration --emitDeclarationOnly \
+                    --outfile ${{ runner.temp }}/schema-pr.d.ts \
+                    packages/perseus-core/src/data-schema.ts
+                  pnpm prettier --write ${{ runner.temp }}/schema-pr.d.ts
+
+            - name: Check out base branch
+              uses: actions/checkout@v4
+              with:
+                  # This checks out the base branch of the PR (the branch the PR
+                  # is targeting)
+                  ref: ${{ github.event.pull_request.base.ref }}
+
+            - name: Re-install node_modules for base branch
+              run: pnpm install
+
+            - name: Build schema for base
+              run: >-
+                  pnpm tsc --declaration --emitDeclarationOnly \
+                    --outfile ${{ runner.temp }}/schema-base.d.ts \
+                    packages/perseus-core/src/data-schema.ts
+
+                  pnpm prettier --write ${{ runner.temp }}/schema-base.d.ts
+
+            - name: Check for Schema Change
+              id: check_for_schema_change
+              run: |
+                  # So a non-zero exit code from 'diff' doesn't bail this step.
+                  set +e
+                  # We generate a random DELIMeter so that the existence of the
+                  # delimeter in the diff output doesn't cause problems.
+                  DELIM=$(uuidgen)
+                  echo "SCHEMA_DIFF<<$DELIM" >> "$GITHUB_OUTPUT"
+                  diff -C 3 ${{ runner.temp }}/schema-base.d.ts ${{ runner.temp }}/schema-pr.d.ts >> "$GITHUB_OUTPUT"
+                  DIFF_EXIT_CODE=$? # Capture the exit code of diff
+                  echo "$DELIM" >> "$GITHUB_OUTPUT"
+                  echo "diff_exit_code=$DIFF_EXIT_CODE" >> "$GITHUB_OUTPUT"
+
+            - name: Schema Changed! Add schema-change label to PR
+              if: steps.check_for_schema_change.outputs.diff_exit_code != '0'
+              env:
+                  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+              run: |
+                  PR_NUMBER="${{ github.event.pull_request.number }}"
+                  gh pr edit "$PR_NUMBER" --add-label "schema-change"
+
+            - name: Schema NOT Changed! Remove schema-change label to PR
+              if: steps.check_for_schema_change.outputs.diff_exit_code == '0'
+              env:
+                  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+              run: |
+                  PR_NUMBER="${{ github.event.pull_request.number }}"
+                  gh pr edit "$PR_NUMBER" --remove-label "schema-change" || true
+
+            - name: Find Schema Change comment
+              uses: peter-evans/find-comment@3eae4d37986fb5a8592848f6a574fdf654e61f9e
+              id: find-schema-change-comment
+              with:
+                  issue-number: ${{ github.event.pull_request.number }}
+                  comment-author: "github-actions[bot]"
+                  body-includes: "Schema Change"
```

Comment:
No action needed, but this seems like it could be a little fragile. I noticed some of our automation does a markdown hack to add a comment to...a comment...I wonder if we could use that to make sure we're editing the right thing.

I don't actually care that much, so if you think this is good enough for now it's good enough for me.

Reply:
This is a frustrating part of these find/add/edit GitHub Actions we use. They don't give you any definitive way to find an existing comment. Once you have the ID your good, but before that you have to find it by text matching.  Perhaps we could put a token at the end of the comment that wouldn't ever be added by a human (like a GUID). But we have this pattern in place in other flows and as far as I know it hasn't failed.